pkg/clusternetwork/hybridOverlay_cni.go

```diff
@@ -0,0 +1,134 @@
+package clusternetwork
+
+import (
+	"encoding/json"
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	wkl "github.com/openshift/windows-machine-config-operator/pkg/controller/wellknownlocations"
+	"github.com/pkg/errors"
+	"io/ioutil"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"os"
+	"path/filepath"
+)
+
+// ovnKubernetes contains information specific to network type OVNKubernetes
+type ovnKubernetes struct {
+	networkType
+	ClusterLevelCniConfig *clusterLevelConfig
+	nodeLevelCniConfig    *NodeLevelConfig
+}
+
+// clusterLevelConfig contains the cluster service CIDR information
+type clusterLevelConfig struct {
+	serviceCIDR string
+}
+
+// NodeLevelConfig contains the node host subnet information
+type NodeLevelConfig struct {
+	hostSubnet string
+}
+
+// cniConfig contains the structure of the cniConfig template
+type cniConfig struct {
+	CniVersion   string `json:"cniVersion"`
+	Name         string `json:"name"`
+	Type         string `json:"type"`
+	Capabilities struct {
```

Comment:
Please dont define a new struct within a different struct. You should move this to outside the struct and define it there. This applies to all instances of this:
```
type capabilities struct {
...
}
```

Reply:
The reason is that if we want to change the value of something like capabilities at a later point we now have to redeclare the definition of the sub-struct like so:
```
	cni := cniConfig{}
	cni.Capabilities = struct {
		Dns bool `json:"dns"`
	}{Dns: true}
```